apis/supernova/src/hooks.js

```diff
@@ -979,3 +980,37 @@ export function onTakeSnapshot(cb) {
   }
   h.fn = cb;
 }
+
+/**
+ * Gets render state instance.
+ *
+ * Used to update properties and get a new layout without triggering onInitialRender.
+ * @entry
+ * @experimental
+ * @returns {{ setPending, restore }} The render state.
+ * @example
+ * import { useRenderState } from '@nebula.js/stardust';
+ *
+ * const renderState = useRenderState();
+ * useState(() => {
+ *   if(needProperteisUpdate(...)) {
+ *      useRenderState.setPending();
+ *      updateProperties(...);
+ *   } else {
+ *      useRenderState.restore();
```

Comment:
It might be a little bit unclear what is actually meant by `restore`... I'm thinking if it would be easier with the following api:
```js
useRenderState.setPending(true);
...
useRenderState.setPending(false);
```

Reply:
yea I kinda agree. But nothing feels right here.

Reply:
should I change it?

if we go with this the renderState object could be removed and the hook return the setPending method
const setPending = useSetPending();
...
setPending(true):
...
setPending(false);
but I unsure if that is more usable.